Cvar VGUI controls constructor changes / general cleanup

braem opened the pull request:
Closes #709 . #710 Is completed and should be closed.

## CvarComboBox
- Changed constructor to not take in a number of items parameter.
    - Number of items is determined by the cvar's min/max values.
- Change all instantiations to use the new constructor.

## CvarTextEntry
- Replaced number format with a precision value (number of decimal points allowed).
    - Better accommodates a corresponding `CvarSlider`.
- Change all instantiations to use the new constructor.
- Cleared up the logic for when to apply changes.
    - Dont apply if last char is a `.`
    - Dont apply if string is all `0`'s
    - Apply when the only char is a `0`, except when the lowest possible value (according to the precision) is not `0`. This allows for editing of `0.X` values without it resetting to the min.

## CvarSlider
- Replaced 'only integers' flag with a precision value much like the one in `CvarTextEntry`.
    - Exposed this to the res files
- Change all instantiations to use the new constructor(s).
- Updated all related res files
- General cleanup on this class

**Other:**
- Set a char limit on `mom_comparisons_max_zones` text entry.
- Removed unnecessary zone menu fields.

### Checklist
- [x] If there is a localization token change, I have updated the `momentum_english_ref.res` file with the changes, ran `tokenizer.py` to generate an up-to-date localization file, and have committed both the `.res` file changes and the new localization `.txt` file
- [x] If I introduced new h/cpp files, I have added them to the appropriate project's VPC file (`server_momentum.vpc` / `client_momentum.vpc` / etc)
- [x] My commits are relatively small and scoped to the best of my ability
- [x] My branch has a clear history of changes that can be easy to follow when being reviewed commit-by-commit
- [x] My branch is functionally complete; the only changes to be done will be those potentially requested in code review

Gocnak commented on mp/src/vgui2/vgui_controls/CvarTextEntry.cpp:
```diff
@@ -40,6 +40,11 @@ void CvarTextEntry::SetPrecision(int precision)
     {
         Q_snprintf(m_szNumberFormat, sizeof(m_szNumberFormat), "%s.%i%s", "%", precision, "f");
     }
+    if (m_cvarRef.GetMin(m_fLowestPossibleVal) && m_fLowestPossibleVal)
```
What happens if the min is negative?

Gocnak commented on mp/src/vgui2/vgui_controls/CvarTextEntry.cpp:
```diff
@@ -180,6 +181,18 @@ bool CvarTextEntry::HasBeenModifiedExternally() const
+bool CvarTextEntry::ShouldUpdate(const char *szText)
+{
+    int strLength = strnlen(szText, MAX_CVAR_TEXT);
+    bool isEqToZero = atof(szText) == 0.0f,               // current text value is 0, 0.0, 0.00, etc
```
```suggestion
    bool isEqToZero = CloseEnough(Q_atof(szText),0.0f, FLT_EPSILON),               // current text value is 0, 0.0, 0.00, etc
```

Gocnak commented on mp/src/vgui2/vgui_controls/CvarTextEntry.cpp:
```diff
@@ -180,6 +181,18 @@ bool CvarTextEntry::HasBeenModifiedExternally() const
+         isJustZero = szText[0] == '0' && strLength == 1, // 1 character and it's 0
+         resetOnZero = m_fLowestPossibleVal == 0.0f;      // if just a 0, dont reset if lowest value is >0
+
+    return (resetOnZero && isJustZero) || !((lastNumIsZero && isEqToZero) || isEndDot);
```
I would just early return these checks as they happen, way more readable that way:

```
bool isEndDot = szText[strLength - 1] = '.';
if (isEndDot)
    return true;

etc
```

braem commented on mp/src/vgui2/vgui_controls/CvarComboBox.cpp:
```diff
@@ -9,29 +9,28 @@ using namespace vgui;
-CvarComboBox::CvarComboBox(Panel *parent, const char *panelName, int numLines, bool allowEdit, char const *cvarname, bool ignoreMissingCvar)
-    : ComboBox(parent, panelName, numLines, allowEdit), m_cvar((cvarname) ? cvarname : "", (cvarname) ? ignoreMissingCvar : true)
+CvarComboBox::CvarComboBox(Panel *parent, const char *panelName, char const *cvarname)
+    : ComboBox(parent, panelName, 5, false), m_cvar(cvarname)
```
> Does this cause a lot of warnings about missing cvars in console?

Nope

> we manually declare them before the res file gets loaded

That's true yeah. Figured it doesn't make much sense to declare this in a res file as items need to be added to it anyhow, which I think can only be done through code. 

> anything initialized via res file (using that `Create_CvarComboBox` guy) will now yell.

Yeah I'm not sure how to get around this given what I assumed above.

Gocnak replied:
> Yeah I'm not sure how to get around this given what I assumed above.

We don't need to, I was just concerned about warnings in this branch, like booting up the game and scoping the console for any, because that points to panels being constructed via res file, which we'd want to fix.

braem replied:
I compared with develop and there are no extra warnings. Searched through the code for references and caught everything by the looks of it.

braem commented:
Implemented the proposed changes and updated with better apply logic.

## CvarSlider
- Added a check on the `CvarSlider` to allow the cvar to be above max or below min if the `usecvarbounds` flag is false

## CvarTextEntry
- Accounted for negative values
- Stopped `-` from being entered when the cvar's min is 0 or above
- Stopped precision of text from getting higher than it should
